htsinfer/get_library_type.py

```diff
@@ -154,47 +165,68 @@ def _align_mates(self):
         concordant = 0
 
         for read1 in samfile1:
-            seq_id1 = read1.query_name
-            if seq_id1 != previous_seq_id1 \
-                    and previous_seq_id1 is not None:
-                mate1.append(reads1.copy())
-                reads1.clear()
-            if read1.reference_end:
-                reads1.append(read1)
+            if not read1.flag & (1 << 2) and isinstance(read1.query_name, str):
```

Comment:
I think this is one of the few cases where I think a comment in the code base would be helpful to read the code. Bitwise operations aren't so common and the multiple conditions further make it a bit hard to follow this.

Reply:
Added comment for clarification.